### PR TITLE
Improved docs for initialization callback

### LINKED ISSUE #409: 401 from Cloudant when require_valid_user=true
Please [read these guidelines](http://ibm.biz/cdt-issue-guide) before opening an issue.

## Bug Description
Two related problems. 

Switching on `require_valid_user=true` has stopped our clients connecting to Cloudant via the client library in this repo. The `POST _session` call always returns a 401, despite the correct username and password being provided (in the url or as separate params). A CURL equivalent works ok and returns 200. Here is the code for our call.

`that.cloudantInit({ url: CDB_URL, plugins: cloudantRequestRetry }, function (err, cloudant) {...})`

Given that CURL works, it must be something the client is doing. Note that we do not provide `cookieauth` as a plugin. However, when we switched on client debug, we could see that the `cookieauth` plugin was **always** being added, whatever value we specified for the `plugins` parameter (a function, empty array, omitted). It turns out that the following line of code is always executed, and is what is adding `cookieauth`...

https://github.com/cloudant/nodejs-cloudant/blob/master/cloudant.js#L300

This is in contradiction to the readme. When that line is disabled, clients connect ok with 200. But are unable to take advantage of session cookies.

So two problems:
1) `cookieauth` is always added, no way to stop that.
2) When `cookieauth` is added, clients get 401 when `require_valid_user=true`.

Catch-22.

We are approaching a code ship deadline, so urgently need a resolution to 1) and preferably 2) as well.

### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue
Set `require_valid_user=true` in Cloudant `.ini` file, start Cloudant, connect using the client, do not specify `cookieauth` plugin.

### 2. What you expected to happen
The client should connect with 200. 

### 3. What actually happened
Client gets 401.

## Environment details
- Version(s) that are affected by this issue.
    > 4.2.2
- Node.js version
    > 10.15

### PR DESCRIPTION
**Pull request by ricellis**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Improve documentation around initialization callback.
Fixes #409

## Approach

Make it clear in the documentation that using the intialization callback will always enable the `cookieauth` plugin.
Provide an example of using the `ping` function as an alternative to the initialization callback

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A doc only changes

## Monitoring and Logging

- "No change"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -249,6 +249,25 @@ Cloudant({ account: me, password: password }, function(err, cloudant, pong) {
...
+
+```js
+const c = new Cloudant({
+  url: 'http://localhost:5984',
+  username: 'somebody',
+  password: 'something',
+  plugins: [] // disable cookieauth
+});
+c.ping()
+  .then(resp => /*do something with response*/ console.log(resp));
```
Comment (glynnbird) [inline README.md]: I don't think you want a semi-colon at the end of this line

Comment (glynnbird) [reply README.md]: Did you know you can do:

```
c.ping().then(console.log).catch(console.error)
```

just to make it less understandable ;) ?

File: README.md [hunk 2]
```diff
@@ -249,6 +249,25 @@ Cloudant({ account: me, password: password }, function(err, cloudant, pong) {
 });
 ~~~
 
+**Note:** For legacy compatibility with older versions of the library using the
+initialization callback will always add the `cookieauth` plugin to the list of
+configured plugins. If you do not want to use `cookieauth` then you should not
+use this initialization callback, instead initalize the client without a
+callback and then call the [ping function](https://console.bluemix.net/docs/services/Cloudant/api/account.html#ping)
```
Comment (glynnbird) [inline README.md]: ping function is now here: https://cloud.ibm.com/docs/services/Cloudant/api?topic=cloudant-account#ping

File: README.md (hunk 1 above)
Comment (ricellis) [reply README.md]: Oof, sloppy.
And neato.
Fixed in 50a96eadab3837d07dfa9aa8dda300a6b5c13006

File: README.md (hunk 2 above)
Comment (ricellis) [reply README.md]: Nice spot, looks like we had a lot of these around still.
Fixed in 1b10733822c893642bbe5cee9d773b623b1742f5

**Pull request merged**